### PR TITLE
add elasticsearch ci

### PR DESCRIPTION
**Pull request by Uzlopak**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by gurgunday — DISMISSED** [review]
lgtm

**Comment by mcollina** [comment]
I'm not sure why we should add this here

**Comment by gurgunday** [comment]
Is it because it's not used often?

Just curious

**Comment by mcollina** [comment]
It's just used for one module. It should stay in that repo.

**Comment by Uzlopak** [comment]
Its just an offer. Probably would remove that github action in the corresponding repo and replace it with the service entry in the github workflow.

**Review by Fdawgs — APPROVED** [review]

**Comment by gurgunday** [comment]
> It's just used for one module. It should stay in that repo.

This is also the case with the [MongoDB](https://github.com/search?q=org%3Afastify+ci-mongo&type=code), [Postgres](https://github.com/search?q=org%3Afastify+ci-postgres&type=code), and [MySQL](https://github.com/search?q=org%3Afastify+ci-mysql&type=code) workflows

Now, following the same logic, it makes sense to move them to the repos where they're used; however, if we merge https://github.com/fastify/workflows/pull/124 and centralize most library CIs too, it could be more efficient in the long term to take it one step further and centralize everything as much as possible

Right now, there is no consistency to when we put a workflow here or to a dedicated repo

Benefits:

- When creating a new repo that will use kafka, mysql, etc., maintainers will know where to look first before thinking about creating a custom workflow
- When the time comes to modify the generic workflow, it will be [easier to also update its flavors](https://github.com/fastify/workflows/pull/116/files)

So I propose that we:

- centralize all workflows that follow [this generic structure](https://github.com/fastify/workflows/blob/main/.github/workflows/plugins-ci.yml) and add one simple thing/feature to it
- only create a dedicated workflow in a repo if it requires significant modifications to the generic workflow

So, in summary, I believe we should merge this and and #113, or remove the other workflows from this repo as well...

**Review by Fdawgs — APPROVED** [review]

**Comment by mcollina** [comment]
I would not have this here, but in the elasticsewrch repo

**Comment by gurgunday** [comment]
Please see the arguments and reasoning at https://github.com/fastify/workflows/pull/114#issuecomment-2067776753

In this repo, there already exists the MongoDB, MySQL, and PostgreSQL workflows that are only used once — in `fastify-mongodb`, `fastify-mysql`, and `fastify-posgres` respectively, we should remove those from here as well then

But I'm arguing that it's actually easier to manage this way, look at https://github.com/fastify/workflows/pull/116 where all it took was one PR to add BlueOak support to the main workflow and all its flavors, this is just another flavor of that [main workflow](https://github.com/fastify/workflows/blob/main/.github/workflows/plugins-ci.yml)

If a repo really requires a custom workflow radically different from the main workflow, then I'm not against putting it in its repo, but in general things will become easier to manage in my opinion

**Review by mcollina — APPROVED** [review]
lgtm
